client/src/modules/events/pages/eventPage.tsx

```diff
@@ -147,9 +134,43 @@ export const EventPage: NextPage = () => {
 
   // TODO: reimplment this the login modal with Auth0
   async function checkOnRsvp(options?: { invited?: boolean }) {
-    await onRsvp(options);
+    const confirmOptions = options?.invited
+      ? {
+          title: 'You have been invited to this event',
+          body: (
+            <>
+              Would you like to attend?
+              <br />
+              Note: joining this event will make you a member of the
+              event&apos;s chapter.
+            </>
+          ),
+        }
+      : {
+          title: 'Join this event?',
+          body: `Note: joining this event will make you a member of the event's chapter.`,
+        };
+    const ok = await confirm(confirmOptions);
+
+    if (ok) {
+      if (!user) {
+        login();
+        modalProps.onOpen();
+        setRsvpAfterLogin(true);
+      } else {
+        await onRsvp();
+      }
+    }
   }
 
+  useEffect(() => {
+    if (rsvpAfterLogin && isLoggedIn) {
+      setRsvpAfterLogin(false);
+      modalProps.onClose();
+      onRsvp();
+    }
+  }, [rsvpAfterLogin, isLoggedIn]);
```

Comment:
I'm guessing this is where the pain came from (or rather that this was how you solved it).

It got me thinking, we could have `login` return the updated meQuery result.  That's probably a good idea in general and would mean you could use the `user` data to get `rsvpStatus` rather than having to wait for the page reload.  How does that sound?

Reply:
It might be useful in general to have that option. Not sure though how convenient it would be to extract such details from it. In this case `rsvp` would need to be included in the meQuery as well.

Reply:
I don't believe so.  We're after `rsvpStatus` which is obtained from `user` (which we don't have yet, but will be able to soon https://github.com/freeCodeCamp/chapter/pull/2009) and from `data` out of `useEventQuery` which is not refetched.

Reply:
Ohh, I totally didn't think about it from this side.